standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -815,26 +855,30 @@ List<Table> getTables(String catName, String dbName, List<String> tableNames, Ge
    * @throws MetaException
    *          Any other errors
    */
-  List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
-      throws MetaException, InvalidOperationException, UnknownDBException, TException;
+  default List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
+      throws MetaException, InvalidOperationException, UnknownDBException, TException{
+    return Collections.emptyList();
+  }
 
   /**
    * Returns the invalidation information for the materialized views given as input.
    */
-  Materialization getMaterializationInvalidationInfo(CreationMetadata cm, String validTxnList)
-          throws MetaException, InvalidOperationException, UnknownDBException, TException;
+  default Materialization getMaterializationInvalidationInfo(CreationMetadata cm, String validTxnList)
+          throws MetaException, InvalidOperationException, UnknownDBException, TException{
+    return new Materialization();
+  }
 
   /**
    * Updates the creation metadata for the materialized view.
    */
-  void updateCreationMetadata(String dbName, String tableName, CreationMetadata cm)
-      throws MetaException, TException;
+  default void updateCreationMetadata(String dbName, String tableName, CreationMetadata cm)
+      throws MetaException, TException{}
 
   /**
    * Updates the creation metadata for the materialized view.
    */
-  void updateCreationMetadata(String catName, String dbName, String tableName, CreationMetadata cm)
-      throws MetaException, TException;
+  default void updateCreationMetadata(String catName, String dbName, String tableName, CreationMetadata cm)
```

Comment:
I am skipping rest, but idea is the same
1. actions should throw `UnsupportedOperationException`
2. we should drop the deprecated API

Reply:
Compared to dropping the deprecated API, I'm wondering if we can introduce a new interface, such as `IMetastoreClientV2`, this class is only for taking care of the lifecycle of the thrift connection. For any table operation, a user can do like this: `TableMetaOperation.of(IMetastoreClientV2, default catalog, configuration).create/drop/getTable(...)`, this is outside of this PR's scope though.

Reply:
I will handle depracation in separate PR.